framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -861,10 +861,11 @@ public AnnotationMirror getAnnotationFromJavaExpression(
     }
 
     /**
-     * Produces the JavaExpression associated with expression on currentPath.
+     * Converts the given {@code expression} string to a {@link JavaExpression} by resolving
+     * identifiers as if they were written in Java code that {@code currentPath}.
      *
      * @param expression a Java expression
-     * @param currentPath the path to an annotation containing {@code expression}
+     * @param currentPath the path to use to resolve identifiers
```

Comment:
What does it mean to resolve an identifier?  It seems to be a different concept than the "resolve" in `resolveMethods`.